### ISSUE TITLE
Fatal warning: No available sequences for HEVC codec extra data

### ISSUE THREAD

**Issue by adamhotep** [peak3d/inputstream.adaptive#774]
When I use the [Netflix add-on](https://github.com/CastagnaIT/plugin.video.netflix) to watch a program encoded with [HEVC](https://en.wikipedia.org/wiki/High_Efficiency_Video_Coding) (H.265), Kodi acknowledges that I'm trying to play something and then gets a pinwheel of death. Waiting does nothing and pressing the "stop" button ("x" on a keyboard) freezes the throbber, but then everything is stuck.

This does not happen with an older program that's encoded with [AVC](https://en.wikipedia.org/wiki/Advanced_Video_Coding) (H.264). (Note, Netflix appears to have a bug in reporting which codec is related to the program. I'm getting the below HEVC warning despite the episode being erroneously marked on the Kodi UI as being H.264 1080 HD with AAC 5.1 audio.)

I'm on an Ubuntu system, so I could ssh in and try `killall -HUP kodi.bin`, but that doesn't work, nor does `killall -STOP kodi.bin`. I've got to go nuclear for `killall -9 kodi.bin`. This tells me something is really stuck. I've got the Netflix add-on set to debugging, but there isn't much there:

```
2021-08-12 22:51:09.150 T:254552    INFO <general>: initializing python engine.
2021-08-12 22:51:09.153 T:254552    INFO <general>: [plugin.video.netflix (24)] Started (Version 1.16.2+matrix.1)
2021-08-12 22:51:09.153 T:254552    INFO <general>: [plugin.video.netflix (24)] URL is ParseResult(scheme='plugin', netloc='plugin.video.netflix', path='/play/show/70281312/season/70281290/episode/70282397/', params='', query='', fragment='')
2021-08-12 22:51:09.376 T:254552    INFO <general>: [plugin.video.netflix (24)] Playing episode_70282397
2021-08-12 22:51:09.605 T:254552 WARNING <general>: [script.module.inputstreamhelper] Widevine update check was made on 2021-08-12 20:50
2021-08-12 22:51:09.896 T:254552    INFO <general>: CPythonInvoker(49, /home/television/.kodi/addons/plugin.video.netflix/addon.py): script successfully run
2021-08-12 22:51:09.904 T:36777    INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.netflix/play/show/70281312/season/70281290/episode/70282397/?profile_guid=0000redacted0000
2021-08-12 22:51:09.904 T:256775    INFO <general>: Creating InputStream
2021-08-12 22:51:09.905 T:256775    INFO <general>: AddOnLog: inputstream.adaptive: SetVideoResolution (1920 x 1080)
2021-08-12 22:51:10.072 T:256777    INFO <general>: [plugin.video.netflix (0)] Requesting licensed manifest for 70282397 with ESN 0000redacted0000 and HDCP []
2021-08-12 22:51:10.073 T:256777    INFO <general>: [plugin.video.netflix (0)] Requested profiles:
                                                   [ 'heaac-2-dash',
                                                     'heaac-2hq-dash',
                                                     'BIF240',
                                                     'BIF320',
                                                     'playready-h264mpl30-dash',
                                                     'playready-h264mpl31-dash',
                                                     'playready-h264mpl40-dash',
                                                     'playready-h264hpl22-dash',
                                                     'playready-h264hpl30-dash',
                                                     'playready-h264hpl31-dash',
                                                     'playready-h264hpl40-dash',
                                                     'webvtt-lssdh-ios8',
                                                     'hevc-main10-L30-dash-cenc',
                                                     'hevc-main10-L31-dash-cenc',
                                                     'hevc-main10-L40-dash-cenc',
                                                     'hevc-main10-L41-dash-cenc',
                                                     'hevc-main10-L50-dash-cenc',
                                                     'hevc-main10-L51-dash-cenc',
                                                     'hevc-main10-L30-dash-cenc-prk',
                                                     'hevc-main10-L31-dash-cenc-prk',
                                                     'hevc-main10-L40-dash-cenc-prk',
                                                     'hevc-main10-L41-dash-cenc-prk',
                                                     'hevc-main10-L30-dash-cenc-prk-do',
                                                     'hevc-main10-L31-dash-cenc-prk-do',
                                                     'hevc-main10-L40-dash-cenc-prk-do',
                                                     'hevc-main10-L41-dash-cenc-prk-do',
                                                     'hevc-main10-L50-dash-cenc-prk-do',
                                                     'hevc-main10-L51-dash-cenc-prk-do']
2021-08-12 22:51:10.293 T:256775    INFO <general>: AddOnLog: inputstream.adaptive: Successfully parsed manifest file. #Periods: 1, #Streams in first period: 6, Type: VOD, Download speed: 9919855.5869 Bytes/s
2021-08-12 22:51:11.020 T:256775    INFO <general>: Creating Demuxer
2021-08-12 22:51:11.020 T:256775    INFO <general>: Opening stream: 1001 source: 256
2021-08-12 22:51:11.071 T:256775 WARNING <general>: AddOnLog: inputstream.adaptive: No available sequences for HEVC codec extra data
2021-08-12 22:51:11.071 T:256775    INFO <general>: Creating video codec with codec id: 173
2021-08-12 22:59:07.296 T:36777    INFO <general>: CVideoPlayer::CloseFile()
2021-08-12 22:59:07.296 T:36777    INFO <general>: VideoPlayer: waiting for threads to exit
```

My Ubuntu 20.04 LTS (Focal) on X86_64 install is fully up to date with a fresh boot and every package is up to date, including kodi 19.1+git20210509.0555-85e05228b4-0\~focal and kodi-inputstream-adaptive 2.6.23-1~focal. Suspecting it might be kernel-related, I upgraded from the latest Focal linux-generic (5.4.0.81.85) to the latest Hirsute linux-generic (5.11.0.25.27), but that doesn't seem to matter. If it's relevant, my VGA controller is listed by `lspci` as "Intel Corporation Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics Controller (rev 06)" on an old Core i5-4440.

I believe the issue is with inputstream.adaptive and/or Widevine (which I've already tried reinstalling a few times via inputstream-helper). Everything worked a few days ago, so I presume this is an artifact of the inputstream.adaptive 2.6.23 release that just came out, but that could have been coincidental with some other update.

**Comment by glennguy** [comment]
Hi @adamhotep 

We need a full debug log: https://kodi.wiki/view/Log_file/Easy

**Comment by adamhotep** [comment]
@glennguy: There appears to be a lot of personal data in the full debug log. I'm concerned about leaking credentials or pii in there.

Here's a [partial dump of my debug-enabled log](https://paste.kodi.tv/ivojoresoz.kodi). It's manually redacted and I've snipped out a lot of content betwen initial Kodi load and launching the Netflix title. Let me know if I've been too aggressive in my redactions and what specifically you're looking for that I've omitted.

**Comment by glennguy** [comment]
Not really sure what it could be. 

@CastagnaIT Any idea on this one? May not be related but is it normal for Netflix to have to use the secure pathway on linux (or any non-Android platform)?

**Comment by CastagnaIT** [comment]
You have enabled the HEVC codec in ntflix addon, if it has not been introduced in the latest widevine versions is not supported by Widevine for linux

For this reason the switch option in the Expert setting of ntfx addon is no available for linux but only for Android,
which makes me think that in some old addon version you have manually enabled HEVC codec or you have forced enabled this codec in some other way

suggest you
1) update the addon to the lastest version
2) in Expert settigns select "Run add-on configuration wizard" to restore default settings

> (Note, Netflix appears to have a bug in reporting which codec is related to the program. I'm getting the below HEVC warning despite the episode being erroneously marked on the Kodi UI as being H.264 1080 HD with AAC 5.1 audio.)

the incorrect media info skins flags, is not an addon bug, is a ntflix website limitation, the website cannot provide the exact info of the videos, because there are too much variables, like device type/type of operative system/type of account/type of SecLev/account settings etc... all these variables influence the video/audio tracks available to the playback (and so the skin info flags are not predeterminable)

**Comment by CastagnaIT** [comment]
> May not be related but is it normal for Netflix to have to use the secure pathway on linux (or any non-Android platform)?

@glennguy i am not sure what you are referring more details please

**Comment by adamhotep** [comment]
@CastagnaIT wrote:
> suggest you
>
> 1. update the addon to the lastest version
> 2. in Expert settigns select "Run add-on configuration wizard" to restore default settings

All of my add-ons are up to date, including Netflix (1.17.0+matrix.1), InputStream Adaptive (2.6.23), and InputStream Heper (0.5.7+matrix.1). I ran the configuration wizard and nothing seemed to change. I don't think I had changed any settings in those three add-ons (or any other add-ons) in the prior months before this, but I did hunt through their settings to see what might be requesting HEVC. I couldn't find anything.

_However_, you gave me the clues I needed to fix this in the XML config file:

```xml
$ cd ~television/.kodi/userdata/addon_data
$ find * -mtime -24 -type f |xargs grep -i hevc
plugin.video.netflix/settings.xml:    <setting id="enable_hevc_profiles">true</setting>
plugin.video.netflix/manifest.json:{"movieId": [70282397], "drmType": "widevine", "type": "standard", "drmVersion": 25,…
plugin.video.netflix/manifest.mpd:<MPD xmlns="urn:mpeg:dash:schema:mpd:2011" xmlns:cenc="urn:mpeg:cenc:2013" mediaPrese…
Binary file script.module.simplecache/simplecache.db matches
```

(To folks trying to research this problem on their own systems, `find * -mtime -24 -type f` finds searches the current directory (including recursively), finding all files modified in the last 24 days. That's because I knew the change happened in that timeframe. Your mileage may vary.)

Given that, I decided to manually edit that file, changing that first hit from `true` to `false`. I did that and restarted Kodi and everything now works. Running that command again gives me different results:
```xml
$ find * -mtime -24 -type f |xargs grep -i hevc
Binary file script.module.simplecache/simplecache.db matches
plugin.video.netflix/settings.xml:    <setting id="enable_hevc_profiles" default="true">false</setting>
```

So it appears the Netflix add-on, at least on my system, thinks HEVC should be enabled by default.

While I now have a functional system, I'm concerned that others may suffer the same issue.

**Comment by glennguy** [comment]
@CastagnaIT was referring to whether Netflix needed to use `SSD_CAPS::SSD_SECURE_PATH` / annex B on non Android platforms, as the log provided has this flag set when the decrypter initializes. Not to worry.

**Comment by adamhotep** [comment]
Glad to hear it. I'm still not sure which setting got tripped on my install (or if the Netflix app should more gracefully fail and perhaps inform the user about the unsupported codec with instructions on how to revert the setting, but that'd make more sense in the Netflix Github issues). It sounds like this ticket is good to close.

**Comment by CastagnaIT** [comment]
that settings for HEVC is not accessible for Linux users
and there is no code in the addon that turn on this settings by itself on linux machine

this is the reason why there are no info in Wiki, because can happen only if a user force something manually or maybe have forced a rollback to a very old version changed manually settings and re-upgrade to last version,
but this back and forth of versions can also cause many others problems

these are all user experiments, i don't worry about that...

**Comment by Emrvb** [comment]
@CastagnaIT 

I started experiencing the same problem since a few weeks, some series just wouldn't play anymore, locking up my media player.
I checked and HEVC profiles were indeed enabled. 

I got this device less than a year ago with a clean install and configuration. The only special thing that happened is that I upgraded from Leia to Matrix.

Is there any chance the toggle for this option is hidden but somehow still accessible/changeble for linux users (invisible but focusable)? Or is there maybe another option that automatically toggles this?

Anyway, keep up the good work.

**Comment by CastagnaIT** [comment]
Kodi devs do not recommend a direct upgrade of Kodi major version
due to possible side effects with the various components
all users that need to do an upgrade of major version should reinstall kodi from scratch
To try fix you can try to do this:

Addon expert settings-->Run addon configuration wizard

if fails reinstall the addon by deleting user data when prompt

**Comment by Emrvb** [comment]
@CastagnaIT 

I already fixed it by editing the settings xml file.

I understand upgrading is not recommended but I would find it odd upgrading would change this setting. Coming to think of it, I've had a similar issue before (pre upgrade) which I resolved by reinstalling the addon. I'll check if I have a backup of the logfile somewhere to see if it's the same issue.

Just consider this a heads up that this problem might not be as rare as you think.
